include/libtorrent/bandwidth_limit.hpp

```diff
@@ -33,30 +33,30 @@ POSSIBILITY OF SUCH DAMAGE.
 #ifndef TORRENT_BANDWIDTH_CHANNEL_HPP_INCLUDED
 #define TORRENT_BANDWIDTH_CHANNEL_HPP_INCLUDED
 
-#include "libtorrent/aux_/disable_warnings_push.hpp"
+#include "libtorrent/assert.hpp"
 
-#include <boost/integer_traits.hpp>
+#include "libtorrent/aux_/disable_warnings_push.hpp"
 #include <boost/cstdint.hpp>
-
 #include "libtorrent/aux_/disable_warnings_pop.hpp"
 
-#include "libtorrent/assert.hpp"
+#include <limits>
 
 namespace libtorrent {
 
 // member of peer_connection
 struct TORRENT_EXTRA_EXPORT bandwidth_channel
 {
-	static const int inf = boost::integer_traits<int>::const_max;
+    enum { inf = std::numeric_limits<boost::int32_t>::max() };
```

Comment:
was this change part of the original patch in master? I don't understand what the purpose of this change is. It seems like a regression. To hold the max value of ``int32_t``, use a ``int32_t`` typed variable.